python/dremio-flight/dremio/arguments/parse.py

```diff
@@ -13,116 +13,118 @@
-    )
-    parser.add_argument(
-        "-user",
-        "--username",
-        type=str,
-        help="Dremio username. Not applicable when connecting to Dremio Cloud",
-        required="-pat" not in sys.argv and "--token" not in sys.argv,
-    )
-    parser.add_argument(
-        "-pass",
-        "--password",
-        type=str,
-        help="Dremio password. Not applicable when connecting to Dremio Cloud",
-        required="-pat" not in sys.argv and "--token" not in sys.argv,
-    )
-    parser.add_argument(
-        "-pat",
-        "--token",
-        dest="token",
-        type=str,
-        help="Either a Personal Access Token or an OAuth2 Token. Only applicable to Dremio Cloud. Use --password if connecting to Dremio Software using PAT",
-        required="-user" not in sys.argv and "--username" not in sys.argv,
-    )
-    parser.add_argument(
-        "-query",
-        "--sqlQuery",
-        dest="query",
-        type=str,
-        help="SQL query to test. Must be enclosed in single quotes. If single quotes are already present within the query, change those to double quotes and enclose entire query in single quotes.",
-        required=True,
-    )
-    parser.add_argument(
-        "-tls",
-        "--tls",
-        dest="tls",
-        help="Enable encrypted connection. Defaults to False.",
-        default=False,
-        action="store_true",
-    )
-    parser.add_argument(
-        "-dcv",
-        "--disableCertificateVerification",
-        dest="disable_certificate_verification",
-        type=bool,
-        help="Disables TLS server verification. Defaults to False.",
-        default=False,
-    )
-    parser.add_argument(
-        "-path_to_certs",
-        "--trustedCertificates",
-        dest="path_to_certs",
+        "-config",
+        "--config-path",
+        dest="config_rel_path",
         type=str,
-        help="Path to trusted certificates for encrypted connection. Defaults to system certificates.",
-        default=certifi.where(),
-    )
-    parser.add_argument(
-        "-sp",
-        "--sessionProperty",
-        dest="session_properties",
-        help="Key value pairs of SessionProperty, example: -sp schema='Samples.\"samples.dremio.com\"' -sp key=value",
-        required=False,
-        nargs="*",
-        action=KVParser,
-    )
-    parser.add_argument(
-        "-engine",
-        "--engine",
-        type=str,
-        help="The specific engine to run against. Only applicable to Dremio Cloud.",
-        required=False,
+        help="Path to config file. Defaults to config.yaml in the same directory.",
+        default="config.yaml",
     )
     return parser.parse_args()
+
+
+def convert_session_properties(session_properties: list[dict]) -> list[tuple]:
+    converted_properties = []
+    for prop in session_properties:
+        converted_properties += [
+            (k.encode("utf-8"), v.encode("utf-8")) for k, v in prop.items()
+        ]
+    return converted_properties
```

Comment:
Arseny also taught me how to apply functional programming in situations like this. If you want to try it as an exercise, here's how you could do it in this case:

```
  return list(map(lambda prop: 
                  list(map(lambda item: (item[0].encode("utf-8"), item[1].encode("utf-8")), prop.items())),
                  session_properties))
```
Here's some good info on how python's map function works

Reply:
I think using a more functional programming approach is just a different approach compared to this more iterative approach, especially when there are no other uses for the functions. However, as a fun exercise, I tried to create a more readable functional approach of this function.

Reply:
Looks great!